flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -363,10 +361,9 @@ public void shutdown(JobStatus jobStatus) throws Exception {
 				masterHooks.clear();
 
 				// clear and discard all pending checkpoints
-				for (PendingCheckpoint pending : pendingCheckpoints.values()) {
-					failPendingCheckpoint(pending, CheckpointFailureReason.CHECKPOINT_COORDINATOR_SHUTDOWN);
-				}
-				pendingCheckpoints.clear();
+				abortPendingCheckpoints(
+					new CheckpointException(
+						CheckpointFailureReason.CHECKPOINT_COORDINATOR_SHUTDOWN));
```

Comment:
it looks like before `triggerQueuedRequests` was not called here (and in some other places), while it's called now?

Reply:
Yes, but `checkAndResetCheckpointScheduler` is called before. They are both trying to resume periodic triggering.
Anyway, `triggerQueuedRequests` will not work here even it's called. Because it will check `shutdown` and `periodicScheduling` first.